### PR TITLE
feat(huey): add Huey Integration

### PR DESCRIPTION
**Pull request by Zhenay**
Add support for the Huey Task Queue https://huey.readthedocs.io/en/latest/

See #195

### CODE REVIEW

**Comment by antonpirker** [comment]
Hey @Zhenay !
We are done reviewing a PyMongo integration and now is the time for Huey! 
I will probably have a look at this later this week

**Comment by j-krl** [comment]
Looking forward to using this!

File: sentry_sdk/integrations/huey.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        with hub.push_scope() as scope:
+            with capture_internal_exceptions():
+                scope._name = "huey"
+                scope.clear_breadcrumbs()
+                scope.add_event_processor(_make_event_processor(task))
+
+            transaction = Transaction(
+                name=task.name,
+                status="ok",
+                op="huey.task",
```
Comment (antonpirker) [inline sentry_sdk/integrations/huey.py:137]: Please change the `op` to be `queue.task.huey`. (see documentation here: https://develop.sentry.dev/sdk/performance/span-operations/)

File: sentry_sdk/integrations/huey.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+    if exc_info[0] in HUEY_CONTROL_FLOW_EXCEPTIONS:
+        hub.scope.transaction.set_status("aborted")
+        return
+
+    hub.scope.transaction.set_status("internal_error")
+    event, hint = event_from_exception(
+        exc_info,
+        client_options=hub.client.options if hub.client else None,
+        mechanism={"type": "huey", "handled": False},
```
Comment (antonpirker) [inline sentry_sdk/integrations/huey.py:92]: Please use here `HueyIntegration.identifier` for setting the type

File: sentry_sdk/integrations/huey.py (hunk 1 above)
Comment (antonpirker) [reply sentry_sdk/integrations/huey.py:137]: Oh and please add `OP.QUEUE_TASK_HUEY` constant here: https://github.com/getsentry/sentry-python/blob/master/sentry_sdk/consts.py#L110 and then use the constant.

File: sentry_sdk/integrations/huey.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    old_enqueue = Huey.enqueue
+
+    def _sentry_enqueue(self, task):
+        # type: (Huey, Task) -> Optional[Union[Result, ResultGroup]]
+        hub = Hub.current
+
+        if hub.get_integration(HueyIntegration) is None:
+            return old_enqueue(self, task)
+
+        with hub.start_span(op="huey.enqueue", description=task.name):
```
Comment (antonpirker) [inline sentry_sdk/integrations/huey.py:52]: Oh and please add `OP.QUEUE_SUBMIT_HUEY` constant (set to "queue.submit.huey") here: https://github.com/getsentry/sentry-python/blob/master/sentry_sdk/consts.py#L110 and then use the constant.

File: tests/integrations/huey/test_huey.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    with start_transaction() as transaction:
+        dummy_task()
+
+    (event,) = events
+
+    assert event["contexts"]["trace"]["trace_id"] == transaction.trace_id
+    assert event["contexts"]["trace"]["span_id"] == transaction.span_id
+
+    assert len(event["spans"])
+    assert event["spans"][0]["op"] == "huey.enqueue"
```
Comment (antonpirker) [inline tests/integrations/huey/test_huey.py:139]: Please keep this a string and do not use the constants (mentioned above) in the tests. This way they tests will fail when the string of the constant changes.

**Review by antonpirker — APPROVED** [review]
Just some cosmetic changes. If those are made and all the tests are green I will create a sample huey setup and give it a spin. (Tomorrow Friday, I am not in front of a computer, so I will be back here on Monday)

**Review by antonpirker — CHANGES_REQUESTED** [review]
Before I clicked wrong. So please make the changes mentioned in the comments in the PR. 

When the changes are made and all the tests are green I will setup a sample huey project and give it a spin. 
(I am not in front of a keyboard tomorrow (Friday) so I will be back here on Monday.)

**Comment by Zhenay** [comment]
Hey @antonpirker 
Could you please run the workflow

**Comment by antonpirker** [comment]
Hey @Zhenay ! I have now set up a simple demo app and everything looks good.

I have the spans for the enqueuing/submitting of tasks:
Screenshot 2023-01-25 at 13 09 38

And also the span for the task itself
Screenshot 2023-01-25 at 13 10 08

I added a little check in your branch to only send sensitive date (the arguments and kwargs of the task) when the option "send_default_pii" is set to true. Because there could be all sorts of sensitive data (passwords, email adresses, names, ... in there) and this data should never be sent by default.

**Review by antonpirker — APPROVED** [review]

**Comment by antonpirker** [comment]
Hey @Zhenay !
Great work, everything looks good. 

In your opinion, is this ready to be merged and shipped?
When there are some bug reports, will you be able to address them?

FYI: The next release will be in around 2 weeks or so.

Oh, and if you send me your shipping address (and t-shirt size) I can see If I can send you some swag as a little thank you. (send it to anton,pirker@sentry.io)

**Comment by Zhenay** [comment]
Hey, @antonpirker!
Thanks for your attention.

I suppose it can be merged. If any bugs are reported, I will be glad to fix them.

**Comment by antonpirker** [comment]
Here you go! Thanks again!

**Comment by ulgens** [comment]
I couldn't find any reference to huey under https://docs.sentry.io/platforms/python/guides/django/configuration/integrations/, is it normal?

**Comment by nick4333** [comment]
Are there any plans to update the documentation to include a huey integration?

**Comment by sentrivana** [comment]
Hey @ulgens, @nick4333, sorry for the late reply, I've created an issue for the missing docs https://github.com/getsentry/sentry-python/issues/2436 and we'll add them shortly!

**Pull request merged**